contentgrid-spring-querydsl/src/main/java/com/contentgrid/spring/querydsl/resolver/CollectionFilterParamPredicateResolver.java

```diff
@@ -33,7 +37,8 @@ public Optional<OperationPredicates> resolve(MethodParameter methodParameter, Cl
         }
 
         return querydslPredicateConverter.getPredicate(domainType, toMultiValueMap(parameters))
-                .map(CollectionFilteringOperationPredicates::new);
+                .<OperationPredicates>map(CollectionFilteringOperationPredicates::new)
+                .or(() -> Optional.of(new CollectionFilteringOperationPredicates(DEFAULT_PREDICATE)));
```

Comment:
Was this needed to make sorting work?

Reply:
Yes. I maybe should have added a comment there.

If there is no predicate present; there is an automatic fallback to the plain RepositoryInvokerAdapter; which does not process `QSort` objects, but processes them as if they are normal `Sort` object (and that fails terribly, because the `Sort.Order` objects are not mapped to java field names at all)